functions

```diff
@@ -148,17 +148,19 @@ kver_x86() {
 }
 
 kver_generic() {
-    # For unknown architectures, we can try to grep the uncompressed
+    # For unknown architectures, we can try to grep the uncompressed or gzipped
     # image for the boot banner.
-    # This should work at least for ARM when run on /boot/Image. On
-    # other architectures it may be worth trying rather than bailing,
-    # and inform the user if none was found.
+    # This should work at least for ARM when run on /boot/Image, or RISC-V on
+    # gzipped /boot/vmlinuz-linuz. On other architectures it may be worth trying
+    # rather than bailing, and inform the user if none was found.
 
     # Loosely grep for `linux_banner`:
     # https://elixir.bootlin.com/linux/v5.7.2/source/init/version.c#L46
-    local kver=
+    local kver= reader=cat
+    
+    [[ $(file -b --mime-type "$1") == 'application/gzip' ]] && reader=zcat
```

Comment:
I'm unsure about the `file` usage. It seems like we use hexdump in `detect_filetype`. We should probably try using that instead of reimplementing this with a new dependency?

Reply:
OK. I have moved the compression detection part out of `detect_filetype` into a new function `detect_compression`, and shared its usage between `lsinitcpio` and `mkinitcpio`.

Reply:
Cool. I would maybe use a case statement for selecting the correct reader thing. But this is better i think :)